src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,72 @@
-     * Given a List of `People` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new People(«Victor», 16, Sex.MAN),
-     * new People(«Helen», 42, Sex.WOMEN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<People> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<People> peopleList) {
-        return Collections.emptyList();
+        Predicate<People> peoplePredicateAgeGender = people ->
+                people.getSex() == People.Sex.MAN
+                && people.getAge() >= fromAge && people.getAge() <= maleToAge
+                || people.getSex() == People.Sex.WOMEN
+                && people.getAge() >= fromAge && people.getAge() <= femaleToAge;
+        return peopleList.stream()
+                .filter(peoplePredicateAgeGender)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `People` instances (having `name`, `age`, `sex` and `List of cats` fields,
-     * and each `Cat` having a `name` and `age`),
-     * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
-     */
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<People> peoplePredicateFemaleAge = people ->
+                people.getSex() == People.Sex.WOMEN
+                && people.getAge() >= femaleAge;
+        return peopleList.stream()
+                .filter(peoplePredicateFemaleAge)
+                .map(People::getCats)
+                .flatMap(Collection::stream)
+                .map(Cat::getName)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Your help with a election is needed. Given list of candidates, where each element
-     * has Candidate.class type.
-     * Check which candidates are eligible to apply for president position and return their
-     * names sorted alphabetically.
-     * The requirements are: person should be older than 35 y, should be allowed to vote,
-     * have nationality - 'Ukrainian'
-     * and live in urk for 10 years. For the last requirement use field periodsInUkr,
-     * which has following view: "2002-2015"
-     * We want to reuse our validation in future, so let's write our own impl of Predicate
-     * parametrized with Candidate in CandidateValidator.
-     */
-    public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+    public static List<String> validateCandidates(List<Candidate> candidates) {
+        return candidates.stream()
+                .filter(candidate -> new CandidateValidator().test(candidate))
```

Comment:
Ми можемо зробити так?
```suggestion
                .filter(new CandidateValidator())
```

Reply:
yeap, fixed